nomad/structs/node.go

```diff
@@ -61,11 +61,18 @@ func (t *CSITopology) Copy() *CSITopology {
 	}
 }
 
-func (t *CSITopology) Equal(o *CSITopology) bool {
+func (t *CSITopology) EqualOrContains(o *CSITopology) bool {
```

Comment:
As shown in the test outputs, this no longer compiles because there are other callers of `Equal`. The `TopologyRequest.Equal` can't use the logic change you've made here either, as it's used to detect validity of changes in the volume specification. So we probably want to break out a `Contains` method from the `Equal` method.

A logic change like this should almost certainly include a change to tests, so once you've got the code compiling let's add that too.

Reply:
Ok, I restored the Equal() function.  Do you want me to add a Unit test for EqualOrContains() method? It's funtionality should be covered by the e2e tests already.

Reply:
> It's funtionality should be covered by the e2e tests already.

While that's true, E2E takes >1hr to run and are challenging to debug, whereas a unit test could give us confidence in any minor changes very cheaply.

I'd also rename this to `Contains`